src/runtime/ArrayBufferObject.cpp

```diff
@@ -70,13 +70,13 @@ ArrayBufferObject::ArrayBufferObject(ExecutionState& state, Object* proto)
     , m_fromExternalMemory(false)
 #endif
 {
-    GC_REGISTER_FINALIZER_NO_ORDER(this, [](void* obj, void*) {
-        ArrayBufferObject* self = (ArrayBufferObject*)obj;
+    addFinalizer([](Object* s, void* data) {
+        ArrayBufferObject* self = s->asArrayBufferObject();
         if (self->m_data) {
```

Comment:
You modified `ArrayBufferObject` finalizer register using `addFinalizer`.
Is this because multiple finalizers could be registered for one ArrayBufferObject?
If so, I think that other Objects which have its own finalizer such as `BigInt` also need to register it with addFinalizer.

Reply:
BTW, we can call `GC_REGISTER_FINALIZER_NO_ORDER` only once for each GC-object?
What if we call multiple `GC_REGISTER_FINALIZER_NO_ORDER` for one object? is it still valid?

Reply:
1. BigInt is not Object. so we can keep GC_REGISTER_FINALIZER_NO_ORDER call
2. we can register only one GC_REGISTER_FINALIZER_NO_ORDER
so, I add {add, remove} finalizer method into Object in order to implement multiple finalizer call

Reply:
Thanks for your kind explanation.
I'll fix WASM*Object finalizer register later since these are Object too.